src/NHibernate.Test/Linq/MiscellaneousTextFixture.cs

```diff
@@ -86,16 +86,17 @@ public void ReferenceToOuter()
 							"the second, third and fourth pages of products")]
 		public void TriplePageSelection()
 		{
-			IQueryable<Product> q = (
-			                        	from p in db.Products
-			                        	where p.ProductId > 1
-			                        	orderby p.ProductId
-			                        	select p
-			);
-
-            IQueryable<Product> page2 = q.Skip(5).Take(5);
-            IQueryable<Product> page3 = q.Skip(10).Take(5);
-            IQueryable<Product> page4 = q.Skip(15).Take(5);
+			var q =
+				from p in db.Products
+				where p.ProductId > 1
+				orderby p.ProductId
+				select p;
+
+			// ToList required otherwise the First call alters the paging and test something else than paging three pages.
```

Comment:
Anyway, I think we shall not change the test. This is a valid linq expression, even if it does not make much sense for you. We shall either skip this test for oracle unchanged or/and create a JIRA issue for a specific case (with a minor/trivial priority).